### PR TITLE
Add `--number-separator` CLI flag

### PR DESCRIPTION
**Pull request by o2sh**
refers to #706 

image
image

Impacted by the CLI flag:

* Number of branches and number of tags
* Number of dependencies
* Number of commits per author
* Number of commits
* LOC
* Number of files

### CODE REVIEW

**Comment by spenserblack** [comment]
Some things to keep in mind:
- [Not all separators are at the thousands place](https://en.wikipedia.org/wiki/Indian_numbering_system#Use_of_separators), so we may get a request to support other groupings, later.
- There are some numerical systems that look like `1.234,5` instead of `1,234.5`.

So, if `--format-numbers` defines the thousands separator, it can be a breaking change if we need to rename that arg to accommodate other options. We could
- rename `--format-numbers` to `--number-separator` or something, or
- change `--format-numbers` totake *names* of formats (we might want to look into what other tools call various number formats).

We could even take a number format like Excel does, `#,###.00`, so that users can define *any* placement of separators, and what types of separators to use, without having to raise future issues. That might be overengineering for our needs, though :laughing:

**Comment by o2sh** [comment]
You're right, my first implementation actually took a localization as an input, but I reverted in 7c6cbba when I found out that [num_crate did not support floating numbers](https://github.com/bcmyers/num-format/issues/1) which would be misleading to users if the decimal character wasn't affected.

I think the `--number-separator` - renamed after your suggestion - CLI flag is a good first step and covers the original issue raised in #706.

File: tests/snapshots/repo__repo.snap [hunk 1]
```diff
@@ -7,8 +7,8 @@ expression: info
   "gitVersion": "git version",
   "project": {
     "repo_name": "repo",
-    "number_of_branches": 0,
-    "number_of_tags": 1
+    "number_of_branches": "0",
+    "number_of_tags": "1"
```
Comment (spenserblack) [inline tests/snapshots/repo__repo.snap:11]: I guess a change in JSON types makes this a breaking change :thinking:

**Review by spenserblack — APPROVED** [review]

File: tests/snapshots/repo__repo.snap (hunk 1 above)
Comment (spenserblack) [reply tests/snapshots/repo__repo.snap:11]: Thinking about this some more, maybe the format should only affect how the stats are *printed*, but keep the struct's types the same, so data outputs don't change their type.

File: tests/snapshots/repo__repo.snap [hunk 2]
```diff
@@ -49,7 +49,7 @@ expression: info
   "lastChange": "22 years ago",
   "contributors": 4,
   "repoUrl": "https://github.com/user/repo.git",
-  "numberOfCommits": "4",
+  "numberOfCommits": 4,
```
[the remark this answers is not in the document]
Comment (spenserblack) [reply tests/snapshots/repo__repo.snap:52]: Yeah, that makes sense to me, too.

**Review by spenserblack — APPROVED** [review]

**Pull request merged**